### PR TITLE
Fixing links (new PR for approved PR #136)

### PR DESCRIPTION
**Pull request by jaret-basden**
# Description

Opening this PR to redo most of the changes in approved [PR #136](https://github.com/apache/polaris/pull/136). PR #136 has gotten out of sync, so I'm opening this new PR instead of rebasing.

- **docs/entities.md**: Updated some links so they use an absolute link. Currently, they use a relative link, which doesn't work with the Redocly output (broken link).
- **docs/overview.md**: Updated some links so they they point to an id attribute to align with the Redocly output. Currently, they point to a raw .md file in the /docs folder. The Redocly output combines .md files in the /docs folder into a single index.html file.
- **docs/quickstart.md**: 
   - Updated some links so they they point to an id attribute to align with the Redocly output. Currently, they point to a raw .md file in the /docs folder. The Redocly output combines .md files in the /docs folder into a single index.html file.
   - Updated ./command-line-interface.md links to https://github.com/polaris-catalog/polaris/blob/main/docs/command-line-interface.md. This exposes a Copy button and presents the GitHub ToC on the left. 

I didn't fix the [docs](./entities.md#privilege) link in docs/command-line-interface.md per [jbonofre's comment in closed PR #231](https://github.com/apache/polaris/pull/231#discussion_r1738704158).

Fixes # (issue): N/A

## Type of change

Please delete options that are not relevant.

- [x] Documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

Please delete options that are not relevant.

- [x] I have tested the links locally in my browser (Chrome)

### CODE REVIEW

**Comment by flyrain** [comment]
I think @eric-maynard's comment in #136 is still valid, https://github.com/apache/polaris/pull/136#discussion_r1715908737

File: docs/index.html [hunk 1]
```diff
@@ -623,10 +623,10 @@ <h3 id="building-polaris">Building Polaris</h3>
...
-&lt;p&gt;Additionally, if Polaris is running somewhere other than &lt;code&gt;localhost:8181&lt;/code&gt;, you can specify the correct hostname and port by providing &lt;code&gt;--host&lt;/code&gt; and &lt;code&gt;--port&lt;/code&gt; flags. For the full set of options supported by the CLI, please refer to the &lt;a href=&quot;./command-line-interface.md&quot;&gt;docs&lt;/a&gt;.&lt;/p&gt;
+&lt;p&gt;If you’re using a storage type other than S3, such as Azure, you’ll provide a different type of credential than a Role ARN. For more details on supported storage types, see the &lt;a href=&quot;#storage-type&quot;&gt;docs&lt;/a&gt;.&lt;/p&gt;
+&lt;p&gt;Additionally, if Polaris is running somewhere other than &lt;code&gt;localhost:8181&lt;/code&gt;, you can specify the correct hostname and port by providing &lt;code&gt;--host&lt;/code&gt; and &lt;code&gt;--port&lt;/code&gt; flags. For the full set of options supported by the CLI, please refer to the &lt;a href=&quot;https://github.com/apache/polaris/blob/main/docs/command-line-interface.md&quot;&gt;docs&lt;/a&gt;.&lt;/p&gt;
```
Comment (eric-maynard) [inline docs/index.html:627]: If I am reading this correctly, clicking a link in a doc on the website will lead you to a doc on github? That feels bad

Comment (flyrain) [reply docs/index.html:627]: To add to it, it points to the main branch only, which breaks other branches.
```
https://github.com/apache/polaris/blob/main/docs/...
```

Comment (flyrain) [reply docs/index.html:627]: I'd recommend to move forward with a way to support a relative paths. We can play with building tools to support it.

File: docs/quickstart.md [hunk 2]
```diff
@@ -158,14 +158,14 @@ This will create a new catalog called **quickstart_catalog**.
...
 
-If you’re using a storage type other than S3, such as Azure, you’ll provide a different type of credential than a Role ARN. For more details on supported storage types, see the [docs](./entities.md#storage-type).
+If you’re using a storage type other than S3, such as Azure, you’ll provide a different type of credential than a Role ARN. For more details on supported storage types, see the [docs](#storage-type).
 
-Additionally, if Polaris is running somewhere other than `localhost:8181`, you can specify the correct hostname and port by providing `--host` and `--port` flags. For the full set of options supported by the CLI, please refer to the [docs](./command-line-interface.md).
+Additionally, if Polaris is running somewhere other than `localhost:8181`, you can specify the correct hostname and port by providing `--host` and `--port` flags. For the full set of options supported by the CLI, please refer to the [docs](https://github.com/apache/polaris/blob/main/docs/command-line-interface.md).
```
Comment (flyrain) [inline docs/quickstart.md:163]: `command-line-interface.md` is in the same directory as this file. We don't need this change.

**Comment by flyrain** [comment]
Hi @jaret-basden ,  the new website is online now, can you double check how this PR applies to new docs framework? I assume some changes are not needed anymore, you can check details here, https://polaris.apache.org/in-dev/unreleased/quickstart/

**Pull request closed without merging**